server/src/test/java/org/elasticsearch/common/unit/RelativeByteSizeValueTests.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.unit;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.test.ESTestCase;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+public class RelativeByteSizeValueTests extends ESTestCase {
+
+    public void testPercentage() {
+        double value = randomIntBetween(0, 100);
+        RelativeByteSizeValue parsed = RelativeByteSizeValue.parseRelativeByteSizeValue(value + "%", "test");
+        assertThat(parsed.getRatio().getAsPercent(), equalTo(value));
+        assertThat(parsed.isAbsolute(), is(false));
+        assertThat(parsed.isNonZeroSize(), is(value != 0.0d));
+    }
+
+    public void testRatio() {
+        double value = (double) randomIntBetween(1, 100) / 100;
+        RelativeByteSizeValue parsed = RelativeByteSizeValue.parseRelativeByteSizeValue(Double.toString(value), "test");
+        assertThat(parsed.getRatio().getAsRatio(),
+            equalTo(value));
```

Comment:
nit: fits on the same line as the previous one

Reply:
Thanks, see 3160c5fb89bbed7d7f77c4c93df5ee2eef952355